design/ceph/submariner-globalnet.md

```diff
@@ -0,0 +1,96 @@
+
+In order to support kubernetes clusters connected using Submariner-Globalnet, Rook needs to use the global IPs provided by Globalnet instead of the service cluster IP.
+
+
+## Prerequisite
+- The peer clusters should be connected using Submariner.
+- Globalnet controller should be running on both the clusters.
+(Refer Submariner [documentation](https://submariner.io/getting-started/quickstart/) for installation steps on different environments)
+
+## Service of OSD pods
+- Implement k8s service for each osd pod:
+```yaml
+apiVersion: v1
+  kind: Service
+  metadata:
+    labels:
+      app: rook-ceph-osd
+      app.kubernetes.io/component: cephclusters.ceph.rook.io
+      app.kubernetes.io/created-by: rook-ceph-operator
+      app.kubernetes.io/instance: <osd-id>
+      app.kubernetes.io/managed-by: rook-ceph-operator
+      app.kubernetes.io/name: ceph-osd
+      app.kubernetes.io/part-of: rook-ceph
+      ceph_daemon_id: <osd-id>
+      ceph_daemon_type: osd
+      osd: <osd-id>
+      osd_cluster: rook-ceph
+      rook.io/operator-namespace: rook-ceph
+      rook_cluster: rook-ceph
+    name: rook-ceph-mon-<id>
+    namespace: rook-ceph
+  spec:
+    selector:
+      app: rook-ceph-osd
+      ceph_daemon_id: <osd-id>
+      osd: <osd-id>
+      osd_cluster: rook-ceph
+      rook_cluster: rook-ceph
+    type: ClusterIP
+```
+## API Changes
+- Add a new `provider` type of `submariner-globalnet` under `spec.network`
+
+``` yaml
+spec:
+    network:
+        ### submariner-globalnet should be used when kubernetes clusters have overlapping CIDRs
+        provider: submariner-globalnet
+```
+
+
+## Service Export
+- `ServiceExport` CR is used to specify which services should be exposed across all the clusters in the cluster set.
+- The exported service becomes accessible as `<service>.<ns>.svc.clusterset.local`
+- Create ServiceExport resource for each mon and osd service.
+```yaml
+apiVersion: multicluster.x-k8s.io/v1alpha1
+  kind: ServiceExport
+    name: <name>
+    namespace: rook-ceph
+```
+- Here, the ServiceExport resource name should be the name of the service that should be exported.
+
+
+
+## Implementation Details
+- Select `submariner-globalnet` as `spec.network.provider`
+- For each mon and osd service:
+    - Create a corresponding `ServiceExport` resource.
+    - Verify the status of the `ServiceExport` from the `ServiceExport` resource status. Sample Status:
+    ```yaml
+    Status:
+    Conditions:
+        Last Transition Time:  2020-12-01T12:35:32Z
+        Message:               Awaiting sync of the ServiceImport to the broker
+        Reason:                AwaitingSync
+        Status:                False
+        Type:                  Valid
+        Last Transition Time:  2020-12-01T12:35:32Z
+        Message:               Service was successfully synced to the broker
+        Reason:
```

Comment:
Why is `Reason` empty here? Isn't this a required field that the export should have?

Reply:
Most probably because its in a `success` state. I'll check again and get back to you on this.